app/models/user.rb

```diff
@@ -29,10 +29,12 @@ class User < ActiveRecord::Base
 
   validates :mobile, format: {with: Mobile::HONGKONGMOBILEREGEXP}, unless: :request_from_stock_without_mobile?
 
-  validates :mobile, presence: true, uniqueness: true, unless: :request_from_stock_without_mobile?
+  validates :mobile, presence: true, unless: :request_from_stock_without_mobile?
+  validates :mobile, uniqueness: true, if: 'mobile.present?'
```

Comment:
Isn't it work without quotes around it? 🤔

Reply:
good point @namrataukirde

Reply:
It works both ways @namrataukirde. With and without quotes too.
=> https://stackoverflow.com/a/20915135